### ISSUE TITLE
Weird DPO loss

### ISSUE THREAD

**Issue by ChenDRAG** [huggingface/alignment-handbook#46]
Hi, I would like to raise some attention to issue #38.

It seems that the DPO-Lora training loss (red line) drops abruptly at the beginning of each epoch, which seems weird.  (I tried Lora model global batch size 64, multi_gpu acceleration, 8GPUs, learning rate 1e-4, others same suggested) 

In the mean time, the full parameter fine tunning has no such problem (official settings). 

![image](https://github.com/huggingface/alignment-handbook/assets/40993476/5ffa7fd5-c93b-44e5-a150-2a133371ab13)

I don't not if this is normal and **assume this is a bug associated with the lora model**. Is there any explanations? Has anyone encountered the same issue? If your rerun loss is normal, can you share your configs?

**Comment by JhonDan1999** [comment]
I am experiencing similar behaviour the training loss values show considerable fluctuations as you can see below 
```
Step	Training Loss
10	1.139900
20	1.149200
30	1.048600
40	1.332400
50	0.924300
60	1.064700
70	0.924300
80	1.140900
90	1.005700
100	1.248100
110	1.067300
120	1.484100
130	2.221100
140	1.040800
150	0.603600
160	1.429100
170	1.215900
180	0.795800
190	1.448500
```

here is my code is there something wrong with the training parameters that caused this behaviour: 
```python

from trl import DPOTrainer


training_arguments = TrainingArguments(
    output_dir="results",
    num_train_epochs=5,
    per_device_train_batch_size=8,
    gradient_accumulation_steps=4,
    optim='paged_adamw_32bit',
    save_steps=10000000000,
    logging_steps=10,
    learning_rate=2e-4,
    weight_decay=2e-4,
    # fp16 = False,  # Set fp16 to False
    bf16=True,
    max_grad_norm=0.3,
    warmup_ratio=0.03,
    lr_scheduler_type='constant',
    save_strategy = "no",


    gradient_checkpointing=True,
    gradient_checkpointing_kwargs={"use_reentrant":False},
    remove_unused_columns=False
)

import os
os.environ['CUDA_LAUNCH_BLOCKING'] = '1'

trainer = DPOTrainer(
    model=peft_model,

    ref_model=None,
    model_init_kwargs=None,
    ref_model_init_kwargs=None,

    tokenizer=tokenizer,
    args=training_arguments,
    beta=0.1,
    loss_type="sigmoid",
    train_dataset=formatted_train_data,
    eval_dataset=None,  # Provide eval dataset if available
    max_length=max_seq_length,
    peft_config=peft_config,

)

trainer.train()
```
